src/main/java/bio/terra/cli/auth/GoogleCredentialUtils.java

```diff
@@ -109,8 +111,9 @@ public String getRedirectUri() {
   private static class NoLaunchBrowser implements AuthorizationCodeInstalledApp.Browser {
     @Override
     public void browse(String url) {
-      System.out.println("Please open the following address in a browser on any machine:");
-      System.out.println("  " + url);
+      PrintWriter out = Printer.getOut();
+      out.println("Please open the following address in a browser on any machine:");
```

Comment:
And do what? For what purpose?

Reply:
To walk through the Google OAuth flow. The default login flow for the CLI is to type `terra auth login` and that launches a browser window on your machine that walks through the Google OAuth flow and sends a code back to the CLI that it can use to get access tokens on your behalf. However, this caused problems on machines that don't have a browser installed by default (e.g. notebook VMs). So, this section of code skips the browser launch and instead just prints out a URL that the user can open in a browser themselves. Importantly, this browser does not need to be on the same machine where they are running the CLI. Then they walk through the OAuth flow and manually copy/paste a code back into the CLI.